### ISSUE TITLE
Set CurrentTime after seeking video?

### ISSUE THREAD

**Issue by colt082295** [videojs/video.js#3159]
I'm loading video piped out in realtime by FFMPEG. My problem is that since it's being passed out in realtime, seeking becomes an issue (I'm currently using webm). When I seek, I spawn another FFMPEG process that seeks to that specific time. This loads the video, but when the video source changes to the new source the time gets set to the start. If I try setting the CurrentTime at the end of the seek, then it gets stuck in a loop.

What's the best way to approach this? Is there any way to set the time in the video without seeking?

    this.on("seeking", function() {

    var time = this.currentTime();

    Meteor.call('streamSeek', location, time, function(error, result) {
    if (error) {

        console.log(error);

    }
    else {

        this.pause()
        this.loadingSpinner.show();
        this.src({
            "type": "video/webm",
            "src": IP + result.port
        });

        this.load();
        this.currentTime(time);

    }

    });

    });

**Comment by gkatsev** [comment]
One thing to try is to wait to call `this.currentTime(time)` after changing the source till after you get the `loadedmetadata` event, otherwise, I'm not quite sure what you're doing or how you have things set up. This question may be more appropriate for a forum like Stack Overflow.

**Comment by colt082295** [comment]
Thanks, sorry for the really late reply. I tried asking the same question on Stack Overflow when I posted the question here, but got no reply.

I'm doing on-the-fly transcoding with ffmpeg, so when the user seeks I have to start a new ffmpeg instance to start transcoding from that specific time the user is seeking to. I'm essentially trying to do what Plex (Plex media server) does with their video player when the user seeks.

The project it's included in is [here](https://github.com/colt082295/Meteor-HTPC).

**Comment by gkatsev** [comment]
Have you tried my suggestion?

**Comment by colt082295** [comment]
Yes, it just starts the seeking event again to send it in an infinite loop.

**Comment by colt082295** [comment]
Either that or I can get it to correctly seek to the time on the seekbar correctly, but the video won't actually start playing until the video has finished transcoding until that point on the seekbar.

The problem being that if I seek to the 20 minute mark in a video, ffmpeg starts transcoding from that point. Then when that video is sent to the browser, it thinks that the 20 minute mark is instead the start of the video (after I set the source to the new transcoding video).
